src/main/java/com/google/sps/data/GoogleShoppingResultsWrapper.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+/**
+ * Return the Google Shopping results for a given query in the form of an HTML page. 
+ */
+public class GoogleShoppingResultsWrapper {
+
+  private static final String GOOGLE_SEARCH_URL = "https://www.google.com//search";
+  private static final String searchType = "tbm=shop";
+  private static final String tbs = "tbs=vw:l"; // "tbs=vw" removes ads and specifies the viewing style.
+  private static final String safetyCheck = "safe=strict";
+
+  public static String getShoppingResultsPage(String shoppingQuery) throws IOException {
+
+    shoppingQuery = shoppingQuery.replaceAll(" ","%20");
```

Comment:
There are several improvements I can see:

#1 this method needs to firstly check whether the input is valid. For example, what will happen if "shoppingQuery = null or empty"?
#2 there should be separate static method to clean the valid input, e.g. trim pre&post empty characters, remove duplicate spaces, replace single space with "%20". For now, this method couldn't handle the following queries:
* "     in       sing   \n"

Please try to avoid assumptions when

Reply:
So, I made sure that shoppingQuery is set to default when no other matching case is found. But just to be double sure I'll make a method to check if its a valid string or not.